lib/puma/single.rb

```diff
@@ -30,13 +30,14 @@ def stop
     end
 
     def halt
-      @server.halt
+      @server.halt if @server
```

Comment:
How does `@server` become nil?

Reply:
Good question.  I saw it used elsewhere, and again, this is 'exit' code.